### PR TITLE
New installation: Configure coreDNS when using kyma local dev domain

### PR DESCRIPTION
**Pull request by tobiscr**
**Description**

Configure coreDNS to properly resolve local Kyma dev domain names.

**Related issue(s)**
#697

### CODE REVIEW

File: cmd/kyma/alpha/deploy/localdomain.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+)
+
+// LocalKymaDevDomain contains the default domain name if Kyma is locally deployed
+const LocalKymaDevDomain = "local.kyma.dev"
+
+// KymaLocalDomain configures the local kyma domain in Kubernetes DNS
+type KymaLocalDomain struct {
+	coreDNSConfig []byte
+	kubeClient    kubernetes.Interface
+}
```
Comment (skhalash) [inline cmd/kyma/alpha/deploy/localdomain.go:18]: In my opinion, the name `KymaLocalDomain` does not really reflect what this type is doing (and is too similar to the constant `LocalKymaDevDomain` above). Maybe something like `LocalDomainDNSConfigurer`?

File: cmd/kyma/alpha/deploy/localdomain.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+// New creates a new Kyma local domain instance
```
Comment (skhalash) [inline cmd/kyma/alpha/deploy/localdomain.go:20]: Could you please document what's exactly happening here?

**Review by skhalash — CHANGES_REQUESTED** [review]
Looks good! Just a few cosmetic comments.

File: cmd/kyma/alpha/deploy/localdomain.go (hunk 1 above)
Comment (tobiscr) [reply cmd/kyma/alpha/deploy/localdomain.go]: Agree, the name is not self-expressive enough. I adjusted it to "DNSConfigurer".

File: cmd/kyma/alpha/deploy/localdomain.go (hunk 2 above)
Comment (tobiscr) [reply cmd/kyma/alpha/deploy/localdomain.go]: Good point, as it's a not self-explaining change. Added more documentation to it.

**Review by skhalash — APPROVED** [review]

File: cmd/kyma/alpha/deploy/dnsconfigurer.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+// NewDNSConfigurer creates a new DNS configurer instance
+func NewDNSConfigurer(kubeClient kubernetes.Interface) *DNSConfigurer {
+	return &DNSConfigurer{
+		kubeClient: kubeClient,
+	}
+}
+
+// ConfigureCoreDNS patches the Kubernetes CoreDNS to support the local kyma domain.
+// It adds the rewrite plugin of CoreDNS to delegate requests to *.local.kyma.dev to the Istio Ingress Gateway.
+func (kld *DNSConfigurer) ConfigureCoreDNS() error {
```
Comment (clebs) [inline cmd/kyma/alpha/deploy/dnsconfigurer.go:33]: Do we really need a type with a single method? Or can this be just a package function?
If there are plans to make it configurable and with different utilities for DNS stuff I could see a type being useful, but at the moment it just adds boilerplate without benefit.

Comment (tobiscr) [reply cmd/kyma/alpha/deploy/dnsconfigurer.go:33]: Good point, I usually prefer types to be a bit more flexible how to handle data. But agree, in this case it unnecessary code.

Comment (tobiscr) [reply cmd/kyma/alpha/deploy/dnsconfigurer.go]: FIXED

Comment (clebs) [reply cmd/kyma/alpha/deploy/dnsconfigurer.go]: It is always possible to turn this function into a method should we need it, but for now it does not hold any data or state.

**Review by clebs — APPROVED** [review]

File: cmd/kyma/alpha/deploy/dnsconfigurer.go (hunk 3 above)
Comment (tobiscr) [reply cmd/kyma/alpha/deploy/dnsconfigurer.go]: agree

**Pull request merged**